Instant-Messenger/src/server/server.cpp

```diff
@@ -198,12 +200,20 @@ namespace server {
 
         // Also, send reference of this instance to the new thread
         args->second = this;
-
+        pthread_create(&monitoringThread, NULL, monitorConnection, (void *) args);
```

Comment:
Pra cada conexão é aberta uma thread de monitoramento de conexão.
Não seria melhor ter apenas 1 thread que monitora todas as conexões?

Reply:
Uma opção. Mas gostei de abrir uma thread para monitorar cada usuário pq assim a gente conta o tempo com base no momento que o usuário conectou.

Reply:
Outra coisa. Abrindo uma thread pra cada a gente meio que deixa um pouco mais genérico pq a gente começa a monitorar um socket. Só precisamos do identificador do socket. Se for fazer um monitoring por uma lista de sockets teria todo mundo que for usar esse monitor vai ter que passar essa lista e chamar a rotina de encerramento de cone'xão pra cada socket que der timeout. Do jeito que tá eu tenho certeza que, quando chegar no join dessa thread `monitoringThread` eu posso numa boa encerrar a conexão pq chegamos no timeout.